File: api/app/models/bet.rb
class Bet < ApplicationRecord
  belongs_to :bettor, class_name: "User", foreign_key: "bettor_id"
  belongs_to :bettee, class_name: "User", foreign_key: "bettee_id"
  belongs_to :winner, class_name: "User", foreign_key: "winner_id"

  belongs_to :game
end

File: app/src/graphql/fragments/bet.js
import gql from 'graphql-tag';
import userFragment from './user';

export default gql`
  ${userFragment}

  fragment betFields on Bet {
    id
    type
    amount
    wager
    bettorPickTeamId
    responded
    accepted
    active
    winnerId
    resolvedAt
    createdAt
    lastRemindedAt
    game {
      id
    }
    bettor {
      ...userFields
    }
    bettee {
      ...userFields
    }
  }
`;
